maplibre-adapter/src/main/java/com/amplifyframework/geo/maplibre/http/AWSRequestSignerInterceptor.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.geo.maplibre.http
+
+import com.amazonaws.DefaultRequest
+import com.amazonaws.http.HttpMethodName
+import com.amazonaws.util.IOUtils
+import com.amplifyframework.geo.location.AWSLocationGeoPlugin
+import okhttp3.*
+
+import okio.Buffer
+import java.io.ByteArrayInputStream
+import java.net.URI
+import java.util.*
+
+private typealias AWSRequest = com.amazonaws.Request<Any>
+
+private fun Request.Builder.copyFrom(request: AWSRequest): Request.Builder {
+    val urlBuilder = HttpUrl.Builder()
+        .host(request.endpoint.host)
+        .scheme(request.endpoint.scheme)
+        .encodedPath(request.encodedUriResourcePath)
+
+    request.parameters.forEach { (name, value) ->
+        urlBuilder.setQueryParameter(name, value)
+    }
+    request.headers.forEach { (name, value) ->
+        this.header(name, value)
+    }
+    return this.url(urlBuilder.build())
+}
+
+/**
+ * Interceptor that can authorize requests using AWS Signature V4 signer.
+ */
+internal class AWSRequestSignerInterceptor(
+    private val plugin: AWSLocationGeoPlugin
```

Comment:
Wouldn't it make more sense to pass the service name directly rather than passing an entire plugin?

Reply:
The signer comes from the plugin itself, in this refactor the dependency is resolved from the Geo category passed to the MapView, which defaults to `Amplify.Geo`.

The interceptor cannot work without the correct plugin, so I think that creating an explicit dependency between both not only favors the actual signer logic (we get whichever signer implementation the SDK provides, even if it changes in the future) but also removes a logic that can fail, where you pass a service name as a string, but your setup is not correct. So although you can sign URLs, the MapView won't work because the plugin hasn't been correctly initialized.